crawlfrontier/contrib/scrapy/schedulers/frontier.py

```diff
@@ -154,4 +170,24 @@ def _get_exception_code(self, exception):
             return '?'
 
     def _request_is_redirected(self, request):
-        return request.meta.get('redirect_times', 0) > 0
\ No newline at end of file
+        return request.meta.get('redirect_times', 0) > 0
+
+    def _add_middlewares(self, crawler):
+        """
+        Adds crawl-frontier scrapy scheduler downloader and spider middlewares.
+        Hack to avoid defining crawl-frontier scrapy middlewares in settings.
```

Comment:
IMO, there is no need to mention that it's a hack to avoid doing something in settings. I think the fact that you inject something automatically in Scrapy needs to be mentioned in documentation.